sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnection.java

```diff
@@ -350,7 +351,11 @@ private synchronized Connection getOrCreateConnection() throws IOException {
             connection = reactor.connectionToHost(handler.getHostname(), handler.getProtocolPort(), handler);
 
             reactorExceptionHandler = new ReactorExceptionHandler();
-            executor = new ReactorExecutor(reactor, Schedulers.single(), connectionId,
+            // Use a new single-threaded scheduler for this connection as QPID's Reactor is not thread-safe.
+            // Using Schedulers.single() will use the same thread for all connections in this process which
+            // limits the scalability of the no. of concurrent connections a single process can have.
+            Scheduler scheduler = Schedulers.newSingle("reactor-executor");
```

Comment:
Since we own this scheduler, we should make sure to dispose of it.

Reply:
@conniey we dispose this scheduler it in the `ReactorExecutor` when [close](https://github.com/Azure/azure-sdk-for-java/pull/17842/files#diff-90c35e5adceef1385767b244fa61d7cce4733e6b7426d2127ae6485d66406230R192) is called.